packages/story-editor/src/components/panels/document/publish/publish.js

```diff
@@ -116,73 +158,115 @@ function PublishPanel() {
-            width: resource.sizes?.full?.width || resource.width,
-            height: resource.sizes?.full?.height || resource.height,
-          },
+  const onNewPublisherLogoSelected = ({ id, src }) => {
+    const newLogo = { id, url: src };
+    addPublisherLogo(id);
+    setPublisherLogos((logos) => [...logos, newLogo]);
+    onPublisherLogoChange(newLogo);
+  };
+
+  const onPublisherLogoChange = (option) => {
+    updateStory({
+      properties: {
+        publisherLogo: {
+          id: option.id,
+          url: option.url,
         },
-      });
-    },
-    [updateStory]
-  );
+      },
+    });
+  };
 
-  const publisherLogoErrorMessage = useMemo(() => {
-    let message = __('No file types are currently supported.', 'web-stories');
+  const getErrorMessage = (message) => {
+    let returnedMessage = __(
+      'No file types are currently supported.',
+      'web-stories'
+    );
 
     if (allowedImageFileTypes.length) {
-      message = sprintf(
-        /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as publisher logo.', 'web-stories'),
+      returnedMessage = sprintf(
+        message,
         translateToExclusiveList(allowedImageFileTypes)
       );
     }
 
-    return message;
-  }, [allowedImageFileTypes]);
+    return returnedMessage;
+  };
 
-  const posterErrorMessage = useMemo(() => {
-    let message = __('No file types are currently supported.', 'web-stories');
+  const publisherLogoErrorMessage = getErrorMessage(
+    /* translators: %s: list of allowed file types. */
+    __('Please choose only %s as publisher logo.', 'web-stories')
+  );
+  const posterErrorMessage = getErrorMessage(
+    /* translators: %s: list of allowed file types. */
+    __('Please choose only %s as a poster.', 'web-stories')
+  );
 
-    if (allowedImageFileTypes.length) {
-      message = sprintf(
-        /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as a poster.', 'web-stories'),
-        translateToExclusiveList(allowedImageFileTypes)
+  const publisherLogoOptionRenderer = forwardRef(
+    ({ option: option, ...rest }, ref) => {
+      if (option.props) {
+        return option;
+      }
+      return (
+        <Option value={option.id} ref={ref} {...rest}>
+          <LogoImg src={option.url} alt="" />
+        </Option>
       );
     }
+  );
+  const activeItemRenderer = () => {
+    const displayText = publisherLogos.length
+      ? __('Select logo', 'web-stories')
+      : __('No logo', 'web-stories');
+    return publisherLogo.id ? (
+      <LogoImg src={publisherLogo.url} alt="" />
+    ) : (
+      <Text as="span" size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+        {displayText}
+      </Text>
+    );
+  };
 
-    return message;
-  }, [allowedImageFileTypes]);
+  const renderUploadButton = (open) => (
+    <Option onClick={open} aria-label={__('Add new', 'web-stories')}>
+      <Icons.ArrowCloud height={32} width={32} />
```

Comment:
Isn't 32x32 the default for icons btw?

Reply:
It will still grow depending on the context, without it: 
![image](https://user-images.githubusercontent.com/21036224/135609814-af5e5c86-a9d6-4b1a-bcc7-eef78ae435d7.png)